src/Components/Aspire.Azure.Messaging.WebPubSub/AspireWebPubSubExtensions.cs

```diff
@@ -74,6 +74,29 @@ public static void AddKeyedAzureWebPubSubServiceClient(
         new WebPubSubComponent().AddClient(builder, DefaultConfigSectionName, configureWithServiceKeyAsDefaultHubName, configureClientBuilder, connectionName: connectionName, serviceKey: serviceKey);
     }
 
+    /// <summary>
+    /// Registers <see cref="WebPubSubServiceClient"/> as a singleton for given <paramref name="connectionName"/> in the services provided by the <paramref name="builder"/>. This
+    /// overload does not require a service key and uses the connection name as the service key
+    /// to support scenarios where multiple Hubs are referenced in the same application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">The name of the component to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureMessagingWebPubSubSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{WebPubSubServiceClient, WebPubSubServiceClientOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire.Azure.Messaging.WebPubSub:{name}" section.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when neither <see cref="AzureMessagingWebPubSubSettings.ConnectionString"/> nor <see cref="AzureMessagingWebPubSubSettings.Endpoint"/> is provided.</exception>
+    public static void AddKeyedAzureWebPubSubServiceClient(
```

Comment:
cc: @vicancy I added an `AddKeyedAzureWebPubSubServiceClient` overload that takes a connection name and uses it as the service key to support scenarios where multiple child hubs are linked and you need to disambiguate between the two.

Let me know what you think of this.